instrumentation/javalin/javalin-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/javalin/v5_0/JavalinTest.java

```diff
@@ -0,0 +1,178 @@
+                            equalTo(ErrorAttributes.ERROR_TYPE, "500"),
+                            equalTo(ClientAttributes.CLIENT_ADDRESS, "127.0.0.1"),
+                            equalTo(NetworkAttributes.NETWORK_PEER_ADDRESS, "127.0.0.1"),
+                            satisfies(
+                                NetworkAttributes.NETWORK_PEER_PORT,
+                                val -> val.isInstanceOf(Long.class)))));
+  }
+
+  @Test
+  public void testSpanNameAndHttpRouteSpanAsyncRouteResponseSuccessful() {
+    client.get("/async").aggregate().join();
+
+    assertNotEquals(port, 0);
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET /async")
+                        .hasKind(SpanKind.SERVER)
+                        .hasNoParent()
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(UrlAttributes.URL_SCHEME, "http"),
+                            equalTo(UrlAttributes.URL_PATH, "/async"),
+                            equalTo(HttpAttributes.HTTP_REQUEST_METHOD, "GET"),
+                            equalTo(HttpAttributes.HTTP_RESPONSE_STATUS_CODE, 200),
+                            satisfies(
+                                UserAgentAttributes.USER_AGENT_ORIGINAL,
+                                val -> val.isInstanceOf(String.class)),
+                            equalTo(HTTP_ROUTE, "/async"),
+                            equalTo(NetworkAttributes.NETWORK_PROTOCOL_VERSION, "1.1"),
+                            equalTo(ServerAttributes.SERVER_ADDRESS, "localhost"),
+                            equalTo(ServerAttributes.SERVER_PORT, port),
+                            equalTo(ClientAttributes.CLIENT_ADDRESS, "127.0.0.1"),
+                            equalTo(NetworkAttributes.NETWORK_PEER_ADDRESS, "127.0.0.1"),
+                            satisfies(
+                                NetworkAttributes.NETWORK_PEER_PORT,
+                                val -> val.isInstanceOf(Long.class)))));
+  }
+
+  @Test
+  void testHttpRouteMetricWithPathParamResponseSuccessful() {
+    String id = "123";
+    AggregatedHttpResponse response = client.get("/param/" + id).aggregate().join();
+    String content = response.contentUtf8();
+    String instrumentation = "io.opentelemetry.jetty-11.0";
+
+    assertNotEquals(port, 0);
+    assertEquals(content, id);
+
+    testing.waitAndAssertMetrics(
+        instrumentation,
+        "http.server.request.duration",
+        metrics ->
+            metrics.anySatisfy(
+                metric -> {
+                  String metricHttpRoute =
+                      metric.getData().getPoints().stream()
+                          .findFirst()
+                          .get()
+                          .getAttributes()
+                          .get(HTTP_ROUTE);
+                  assertEquals("/param/{id}", metricHttpRoute);
```

Comment:
Usually we don't assert the route in metrics. If the route in span is as expected then the metics should also be fine. I think it would be more conventional to write this assertion as follows
```
assertThat(metric).hasHistogramSatisfying(
  histogram ->
    histogram.hasPointsSatisfying(
      point ->
        point.hasAttribute(HTTP_ROUTE, "/param/{id}")
```

Reply:
While our reliance on `HttpServerRoute.update` might suggest that we need not concern ourselves with the `http.route` attribute not matching the `span name`, I've included it for a reason. In various attempts to fix the issue, I encountered scenarios where this assumption did not hold true. For instance, this discrepancy arose when calling `HttpServerRoute.update` with a `HttpServerRouteSource` other than `HttpServerRouteSource.CONTROLLER`.

I'd like to keep this assertion, just in case, and I've changed it to your suggested implementation and looks so much better, thanks 😄